Vitest 4.x internal typescript error with Vite <=7.1.0

### Describe the bug

Vitest 4.x lists `"vite": "^6.0.0 || ^7.0.0"` as a dependency, but if you use any version of `vite` before v7.1.0, TypeScript errors with:

```
node_modules/vitest/dist/chunks/moduleRunner.d.BxT-OjLR.d.ts:174:74 - error TS2339: Property 'createImportMeta' does not exist on type 'ModuleRunnerOptions'.

174 type CreateImportMeta = NonNullable<viteModuleRunner.ModuleRunnerOptions["createImportMeta"]>;
                                                                             ~~~~~~~~~~~~~~~~~~
```

Note that this only happens if the `vitest.config.ts` file is included in the TypeScript context.

### Reproduction

https://stackblitz.com/edit/vitest-dev-vitest-lew46gxc?file=tsconfig.json

### System Info

```shell
System:
    OS: macOS 15.7.2
    CPU: (14) arm64 Apple M4 Pro
    Memory: 524.47 MB / 48.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 24.11.1 - /Users/<redacted>/.nvm/versions/node/v24.11.1/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 11.6.2 - /Users/<redacted>/.nvm/versions/node/v24.11.1/bin/npm
    Watchman: 2025.04.21.00 - /run/current-system/sw/bin/watchman
  Browsers:
    Firefox: 145.0
    Firefox Developer Edition: 145.0
    Safari: 26.1
  npmPackages:
    vite: 7.0.4 => 7.0.4 
    vitest: 4.0.10 => 4.0.10
```

### Used Package Manager

npm

### Validations

- [x] Follow our [Code of Conduct](https://github.com/vitest-dev/vitest/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guidelines](https://github.com/vitest-dev/vitest/blob/main/CONTRIBUTING.md).
- [x] Read the [docs](https://vitest.dev/guide/).
- [x] Check that there isn't [already an issue](https://github.com/vitest-dev/vitest/issues) that reports the same bug to avoid creating a duplicate.
- [x] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/vitest-dev/vitest/discussions) or join our [Discord Chat Server](https://chat.vitest.dev).
- [x] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

For me it is a bit more of a blocker, because I don't have a direct `vite` dependency anywhere, only `vitest`. Force-upgrading `npm update vite` (6.3.5 -> 7.2.4) also doesn't help: while it solves the error OP described, I also had a second error:

```
node_modules/vitest/dist/chunks/moduleRunner.d.RBEiFdiW.d.ts:14:28 - error TS2315: Type 'Buffer' is not generic.

14  readBuffer(path: string): Buffer<ArrayBuffer>;
```

and that still remains (no idea what's causing it -- could be a node thing (I'm on 25))

I can get `tsc` to finally behave with the `skipLibCheck: true` option (regardless of what version vite is at) but it's subideal. "Temporary" often means "forever" on corporate projects.

It also feels kinda wrong to have to make that setting less strict to get the new _much stricter_ coverage stuff (it's already poking holes in files that _used to be_ reported as 100%). I've seen enough times where these kind of typescirpt errors basically force people to use somewhat decent dependencies that I don't want to "temporary" suppress it.

--- a/packages/vitest/src/runtime/moduleRunner/moduleRunner.ts
+++ b/packages/vitest/src/runtime/moduleRunner/moduleRunner.ts
@@ -11,7 +11,7 @@ import { Traces } from '../../utils/traces'
 import { VitestMocker } from './moduleMocker'
 import { VitestTransport } from './moduleTransport'
 
-export type CreateImportMeta = NonNullable<viteModuleRunner.ModuleRunnerOptions['createImportMeta']>
+export type CreateImportMeta = (modulePath: string) => viteModuleRunner.ModuleRunnerImportMeta | Promise<viteModuleRunner.ModuleRunnerImportMeta>
 export const createNodeImportMeta: CreateImportMeta = (modulePath: string) => {
   if (!viteModuleRunner.createDefaultImportMeta) {
     throw new Error(`createNodeImportMeta is not supported in this version of Vite.`)

